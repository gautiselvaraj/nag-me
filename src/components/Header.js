import React from 'react';
import styled from 'styled-components';
import H1 from './H1';
import Button from './Button';

const Header = styled.header`
  align-items: center;
  background-color: ${props => props.theme.greyLightest};
  border-bottom: 1px solid ${props=> props.theme.greyLighter};
  display: flex;
  padding: 10px;
`;

const Heading = styled.div`
  flex-grow: 1;
`;

const Link = styled.div`
  flex: 0 0 20px;
`;

export default () => (
  <Header>
    <Heading>
      <H1>Nag Me</H1>
    </Heading>
    <Link>
      <Button link circle href="https://www.gauti.info/nag-me" target="_blank" title="More information about Nag Me">
        ?
      </Button>
    </Link>
  </Header>
);
