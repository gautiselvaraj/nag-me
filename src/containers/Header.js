import {connect} from 'react-redux';
import Header from '../components/Header';
import {nagIndex, nagNew} from '../actions/NagActions';

const mapDispatchToProps = dispatch => ({
  nagIndex: () => dispatch(nagIndex()),
  nagNew: () => dispatch(nagNew())
});

export default connect(null, mapDispatchToProps)(Header);
