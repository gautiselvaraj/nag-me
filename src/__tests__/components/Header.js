import React from 'react';
import { mount } from 'enzyme';
import Header from '../../components/Header';
import 'jest-styled-components';

describe('<Header />', () => {
  let nagIndex, nagNew, header;

  beforeEach(() => {
    nagIndex = jest.fn();
    nagNew = jest.fn();
    header = mount(<Header nagIndex={nagIndex} nagNew={nagNew} />);
  });

  it('should match snapshot and have correct elements and children', () => {
    expect(header).toMatchSnapshot();
    expect(header.find('header').length).toBe(1);
    expect(header.find('a[href="https://www.gauti.info/nag-me"]').length).toBe(
      1
    );
    expect(header.find('button[title="Nags Index"]').length).toBe(1);
    expect(header.find('button[title="Create a nag"]').length).toBe(1);
  });

  it('should call nagIndex', () => {
    header.find('button[title="Nags Index"]').simulate('click');
    expect(nagIndex.mock.calls.length).toBe(1);
  });

  it('should call nagNew', () => {
    header.find('button[title="Create a nag"]').simulate('click');
    expect(nagNew.mock.calls.length).toBe(1);
  });
});
