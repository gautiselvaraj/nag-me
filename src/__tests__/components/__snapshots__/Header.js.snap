// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot and have correct elements and children 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
}

.c1 {
  -webkit-flex: 0 0 25px;
  -ms-flex: 0 0 25px;
  flex: 0 0 25px;
  height: 18px;
  text-align: center;
}

.c2 {
  text-decoration: none;
}

.c3 {
  color: undefined;
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c3:before {
  content: "\\e901";
}

.c7 {
  color: undefined;
  font-family: icons;
  font-size: 1rem;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  line-height: 1;
  speak: none;
  text-transform: none;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c7:before {
  content: "\\ea09";
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: center;
}

.c5 {
  background-color: transparent;
  border: 0;
  cursor: pointer;
  display: inline-block;
  padding: 0;
}

.c5:hover {
  background-color: transparent;
}

.c6 {
  font-size: 1.25rem;
  margin-top: 0;
  margin-bottom: 0;
}

<Header
  nagIndex={[Function]}
  nagNew={[Function]}
>
  <styled.header>
    <header
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <Component
            href="https://www.gauti.info/nag-me"
            target="_blank"
            title="More information about Nag Me"
          >
            <styled.a
              href="https://www.gauti.info/nag-me"
              target="_blank"
              title="More information about Nag Me"
            >
              <a
                className="c2"
                href="https://www.gauti.info/nag-me"
                target="_blank"
                title="More information about Nag Me"
              >
                <Component
                  inverse={true}
                  question={true}
                >
                  <styled.i
                    inverse={true}
                    question={true}
                  >
                    <i
                      className="c3"
                    />
                  </styled.i>
                </Component>
              </a>
            </styled.a>
          </Component>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          <Component
            onClick={[Function]}
            reset={true}
            title="Nags Index"
          >
            <styled.button
              onClick={[Function]}
              reset={true}
              title="Nags Index"
            >
              <button
                className="c5"
                onClick={[Function]}
                title="Nags Index"
              >
                <Component>
                  <styled.h1>
                    <h1
                      className="c6"
                    >
                      Nag Me
                    </h1>
                  </styled.h1>
                </Component>
              </button>
            </styled.button>
          </Component>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="c1"
        >
          <Component
            onClick={[Function]}
            reset={true}
            title="Create a nag"
          >
            <styled.button
              onClick={[Function]}
              reset={true}
              title="Create a nag"
            >
              <button
                className="c5"
                onClick={[Function]}
                title="Create a nag"
              >
                <Component
                  add={true}
                  inverse={true}
                >
                  <styled.i
                    add={true}
                    inverse={true}
                  >
                    <i
                      className="c7"
                    />
                  </styled.i>
                </Component>
              </button>
            </styled.button>
          </Component>
        </div>
      </styled.div>
    </header>
  </styled.header>
</Header>
`;
